File: internal/models/deploy/nacos_release.go
package deploy

import "time"

// NacosRelease Nacos 配置发布单
type NacosRelease struct {
	ID               uint       `gorm:"primarykey" json:"id"`
	CreatedAt        time.Time  `json:"created_at"`
	UpdatedAt        time.Time  `json:"updated_at"`
	Title            string     `gorm:"size:200;not null" json:"title"`
	NacosInstanceID  uint       `gorm:"not null;index" json:"nacos_instance_id"`
	NacosInstanceName string    `gorm:"size:100" json:"nacos_instance_name"`
	Tenant           string     `gorm:"size:200" json:"tenant"`
	Group            string     `gorm:"size:200;not null;default:'DEFAULT_GROUP'" json:"group"`
	DataID           string     `gorm:"size:200;not null" json:"data_id"`
	Env              string     `gorm:"size:30;not null;default:'dev'" json:"env"`
	ConfigType       string     `gorm:"size:20;default:'yaml'" json:"config_type"`
	ContentBefore    string     `gorm:"type:longtext" json:"content_before"`
	ContentAfter     string     `gorm:"type:longtext" json:"content_after"`
	ContentHash      string     `gorm:"size:64" json:"content_hash"`
	ServiceID        *uint      `gorm:"index" json:"service_id"`
	ServiceName      string     `gorm:"size:100" json:"service_name"`
	ReleaseID        *uint      `gorm:"index" json:"release_id"`
	Status           string     `gorm:"size:20;not null;default:'draft';index" json:"status"`
	RiskLevel        string     `gorm:"size:20;default:'low'" json:"risk_level"`
	Description      string     `gorm:"type:text" json:"description"`
	CreatedBy        uint       `json:"created_by"`
	CreatedByName    string     `gorm:"size:100" json:"created_by_name"`
	ApprovedBy       *uint      `json:"approved_by"`
	ApprovedByName   string     `gorm:"size:100" json:"approved_by_name"`
	ApprovedAt       *time.Time `json:"approved_at"`
	ApprovalInstanceID *uint    `gorm:"index" json:"approval_instance_id,omitempty"`
	ApprovalChainID    *uint    `gorm:"index" json:"approval_chain_id,omitempty"`
	ApprovalChainName  string   `gorm:"size:100" json:"approval_chain_name"`
	PublishedAt      *time.Time `json:"published_at"`
	PublishedBy      *uint      `json:"published_by"`
	PublishedByName  string     `gorm:"size:100" json:"published_by_name"`
	RollbackFromID   *uint      `json:"rollback_from_id"`
	RejectReason     string     `gorm:"size:500" json:"reject_reason"`
}

// Status: draft -> pending_approval -> approved -> published -> rolled_back
//         draft -> pending_approval -> rejected

func (NacosRelease) TableName() string { return "nacos_releases" }

File: web/src/services/nacosRelease.ts
import request from './api'
import type { ApiResponse, PageResponse } from '../types'

export interface NacosRelease {
  id?: number
  created_at?: string
  updated_at?: string
  title: string
  nacos_instance_id: number
  nacos_instance_name: string
  tenant: string
  group: string
  data_id: string
  env: string
  config_type: string
  content_before: string
  content_after: string
  content_hash: string
  service_id?: number
  service_name: string
  release_id?: number
  status: string
  risk_level: string
  description: string
  created_by: number
  created_by_name: string
  approved_by?: number
  approved_by_name: string
  approved_at?: string
  approval_instance_id?: number
  approval_chain_id?: number
  approval_chain_name?: string
  published_at?: string
  published_by?: number
  published_by_name: string
  rollback_from_id?: number
  reject_reason: string
}

export interface NacosReleaseFilter {
  env?: string
  status?: string
  data_id?: string
  service_id?: number
  page?: number
  page_size?: number
}

export const nacosReleaseApi = {
  list(params: NacosReleaseFilter) {
    return request.get<PageResponse<NacosRelease>>('/nacos-releases', { params })
  },
  getById(id: number) {
    return request.get<ApiResponse<NacosRelease>>(`/nacos-releases/${id}`)
  },
  getByApprovalInstance(approvalInstanceId: number) {
    return request.get<ApiResponse<NacosRelease>>(`/nacos-releases/by-approval/${approvalInstanceId}`)
  },
  create(data: Partial<NacosRelease>) {
    return request.post<ApiResponse<NacosRelease>>('/nacos-releases', data)
  },
  update(id: number, data: Partial<NacosRelease>) {
    return request.put<ApiResponse<NacosRelease>>(`/nacos-releases/${id}`, data)
  },
  delete(id: number) {
    return request.delete<ApiResponse<void>>(`/nacos-releases/${id}`)
  },
  submit(id: number) {
    return request.post<ApiResponse<NacosRelease>>(`/nacos-releases/${id}/submit`)
  },
  approve(id: number) {
    return request.post<ApiResponse<NacosRelease>>(`/nacos-releases/${id}/approve`)
  },
  reject(id: number, reason: string) {
    return request.post<ApiResponse<NacosRelease>>(`/nacos-releases/${id}/reject`, { reason })
  },
  publish(id: number) {
    return request.post<ApiResponse<NacosRelease>>(`/nacos-releases/${id}/publish`)
  },
  rollback(id: number) {
    return request.post<ApiResponse<NacosRelease>>(`/nacos-releases/${id}/rollback`)
  },
  fetchContent(params: { instance_id: number; tenant: string; group: string; data_id: string }) {
    return request.get<ApiResponse<string>>('/nacos-releases/fetch-content', { params })
  },
  listByService(serviceId: number, limit?: number) {
    return request.get<ApiResponse<NacosRelease[]>>(`/nacos-releases/by-service/${serviceId}`, { params: { limit } })
  },
}
